kernel/src/log_compaction/mod.rs

```diff
@@ -0,0 +1,108 @@
+//! This module provides an API for writing log compaction files that aggregate
+//! multiple commit JSON files into single compacted files. This improves performance
+//! by reducing the number of individual log files that need to be processed during
+//! log replay operations.
+//!
+//! ## Overview
+//!
+//! Log compaction creates files with the naming pattern `{start_version}.{end_version}.compacted.json`
+//! that contain the reconciled actions from all commit files in the specified version range.
+//! This is similar to checkpoints but operates on a subset of versions rather than the entire table.
+//!
+//! ## Usage
+//!
+//! The log compaction API follows a similar pattern to the checkpoint API:
+//!
+//! 1. Create a [`LogCompactionWriter`] using [`Snapshot::compact_log`]
+//! 2. Get the compaction path from [`LogCompactionWriter::compaction_path`]
+//! 3. Get the compaction data from [`LogCompactionWriter::compaction_data`]
+//! 4. Write the data to the path in object storage (engine-specific)
```

Comment:
this design certainly seems reasonable but given that we already have facilities for writing JSON (via `Engine` trait) I wonder if log compaction would make sense to just write out "ourselves" (kernel writes via Engine trait instead of the more complex choreography for checkpointing)

that probably only makes sense if we know that the compaction will be small, so perhaps this approach is generally superior but figured i'd raise!

Reply:
Compaction results can be quite large (depends on interval and size of commits in the interval), so I thought of letting the engine write it themselves made sense. Happy to change it if it is overly complicated.